sdk/storage/azure-storage-files-datalake/inc/azure/storage/files/datalake/datalake_directory_client.hpp

```diff
@@ -200,6 +241,14 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
         : DataLakePathClient(std::move(dfsUrl), std::move(blobClient), pipeline)
     {
     }
+
+    Azure::Core::Response<Models::SetDataLakeDirectoryAccessControlRecursiveSinglePageResult>
+    SetAccessControlRecursiveSinglePageInternal(
+        Models::PathSetAccessControlRecursiveMode mode,
+        std::vector<Models::Acl> acls,
+        const SetDataLakeDirectoryAccessControlRecursiveSinglePageOptions& options
+        = SetDataLakeDirectoryAccessControlRecursiveSinglePageOptions()) const;
+
```

Comment:
I suggest you move this function to .cpp file to make this header file clearer

Reply:
Then I need to add 3 API implementations in cpp file. I think it is only a personal preference so I'll just leave it here.

Reply:
okay, whatever.